workflow_plandefinition/models/workflow_plan_definition_action.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services, S.L.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import fields, models, api, exceptions
+
+
+class PlanDefinitionAction(models.Model):
+    _name = 'workflow.plan.definition.action'
+    _description = 'Medical Plan Definition Action'
+    _inherit = 'mail.thread'
+    _parent_name = 'parent_id'
+    _parent_store = True
+    _parent_order = 'name'
+    _order = 'parent_left'
+    _rec_name = 'complete_name'
+    _order = 'id asc'
+
+    @api.one
+    @api.depends('name', 'parent_id')
+    def _compute_complete_name(self):
+        """ Forms complete name of action from parent to child action. """
+        name = self.name
+        current = self
+        while current.parent_id:
+            current = current.parent_id
+            name = '%s/%s' % (current.name, name)
+        self.complete_name = name
+
+    @api.multi
+    @api.depends('parent_id', 'direct_plan_definition_id')
+    def _compute_plan_definition_id(self):
+        for rec in self:
+            rec.plan_definition_id = rec.direct_plan_definition_id
+            if rec.parent_id:
+                rec.plan_definition_id = rec.parent_id.plan_definition_id
+
+    name = fields.Char(
+        string='Action name',
```

Comment:
Stylistic, but IMO `Action` is implied in all of these and is superfluous. Our current context is a plan definition action, so it's implicit

Reply:
removed string and understood the message! :+1: